packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -202,15 +204,45 @@ export default class InternalModel {
     // `lastObject` have changed.  When this happens we don't want those
     // models to rematerialize their records.
 
+    // eager checks to avoid instantiating record data if we are empty or loading
+    if (this.isEmpty()) {
+      return true;
+    }
+
+    if (this.isLoading()) {
```

Comment:
this seems to be a new addition to our logic?

Reply:
I flagged it, but the addition is due to `_isRecordFullyDeleted` accessing recordData, which if you are `isLoading` means that you will eagerly instantiate the RD, even though we know none of the other properties should be true. Great call on flagging though, as it turns out right now all of the tests pass both when returning false and when returning true. I will make sure to add a test on master that asserts the correct behavior